### PR TITLE
style: Regularize lint

### PR DESCRIPTION
**Pull request by kriskowal**
This change makes the `eslintConfig` block the same in every package. A new `eslint-config-template.json` appears at the top-level with a `scripts/sync-eslint-config.sh` that will push this block into every package’s `package.json` centrally. This should eventually be replaced with a lint config package.

This subsumes many global declarations common to the Agoric execution environment and removes the explicit global hints from every file that previously had them.

The consolidated configuration requires valid jsdocs. To get lint passing, some are fixed, others are demoted to normal comments where the gap is too vast and ts-check is not yet enabled.

One file had DOS line-endings, which will now be UNIX line-endings.

### CODE REVIEW

File: packages/eslint-config/eslint-config.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    "StaticModuleRecord": "readonly"
+  },
+  "rules": {
+    "implicit-arrow-linebreak": "off",
+    "function-paren-newline": "off",
+    "arrow-parens": "off",
+    "strict": "off",
+    "prefer-destructuring": "off",
+    "no-else-return": "off",
+    "no-console": "off",
```
Comment (erights) [inline packages/eslint-config/eslint-config.json]: And this makes the implicit inclusion of `"console"` doubly confusing. Do they by default both include `"console"` and prohibit it?

File: packages/eslint-config/eslint-config.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  "extends": [
+    "airbnb-base",
+    "plugin:prettier/recommended",
+    "plugin:jsdoc/recommended"
+  ],
+  "parser": "@typescript-eslint/parser",
+  "env": {
+    "es6": true
+  },
+  "globals": {
```
Comment (erights) [inline packages/eslint-config/eslint-config.json]: Should `"HandledPromise"` and/or `"E"` be on this list?

File: packages/sparse-ints/src/sparseInts.js [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
+// Copyright (C) 2019 Agoric, under Apache License 2.0
+
+/**
+ * Generator function to produce a stream of positive integers that are
+ * sparsely scattered across the number space. This supports IDs that
+ * are guessable, but for example mistyping a correct ID is unlikely to
+ * mistakenly match another generated ID.
+ *
+ * @param {number} seed
+ * @yields {number}
```
Comment (erights) [inline packages/sparse-ints/src/sparseInts.js:10]: Wow. I didn't know jsdoc had a `@yields`. Cool!

File: packages/spawner/src/contractHost.js [hunk 4]
```diff
@@ -10,7 +8,7 @@ import { makeIssuerKit } from '@agoric/ertp';
 
 export { makeCollect } from './makeCollect';
 
-/**
+/*
```
Comment (erights) [inline packages/spawner/src/contractHost.js:11]: Why remove the second star?

File: packages/spawner/src/escrow.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 // Copyright (C) 2019 Agoric, under Apache License 2.0
 
-/* global harden E makePromiseKit */
+/* global E makePromiseKit */
```
Comment (erights) [inline packages/spawner/src/escrow.js:3]: Who makes `makePromiseKit` a global? Why?

This need not be answered in this PR.

File: packages/swingset-runner/demo/resolveChain/bootstrap.js [hunk 6]
```diff
@@ -9,7 +9,7 @@ export function buildRootObject(_vatPowers, options) {
   function waitFor(who, p) {
     p.then(
       answer => {
-        if (0 < count && count < 50) {
+        if (count > 0 && count < 50) {
```
Comment (erights) [inline packages/swingset-runner/demo/resolveChain/bootstrap.js:12]: This bug fix is completely orthogonal to this PR, yes? Was it a known bug?

**Review by erights — APPROVED** [review]
Awesome thanks for cleaning this up! Truly!

LGTM

File: packages/eslint-config/eslint-config.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (kriskowal) [reply packages/eslint-config/eslint-config.json]: I’m not sure where it comes from. I couldn’t find an answer in `eslint-plugin-airbnb-base`. It might be assumed by `eslint` proper.

File: packages/eslint-config/eslint-config.json (hunk 1 above)
Comment (kriskowal) [reply packages/eslint-config/eslint-config.json]: That is likely. The difference would likely be in the shape of the warning or error, from “I don’t know about that” to “I know about that but you shouldn’t use it”

File: packages/eslint-config/eslint-config.json (hunk 2 above)
Comment (kriskowal) [reply packages/eslint-config/eslint-config.json]: They are not currently necessary to pass tests today and they could be rolled up.

File: packages/sparse-ints/src/sparseInts.js (hunk 3 above)
Comment (kriskowal) [reply packages/sparse-ints/src/sparseInts.js:10]: This turns out to not just be cool but also necessary to pass the jsdoc eslint rules. They appear to take a dim view on the equivalent `returns {Generator<T, U, V>}`.

File: packages/spawner/src/contractHost.js (hunk 4 above)
Comment (kriskowal) [reply packages/spawner/src/contractHost.js:11]: This is not a valid jsdoc, since it contains commentary but doesn’t declare the signature. The file doesn’t have `@ts-check` so it’s okay for it to be informal, but in normalizing the lint rules, I entrained the `jsdoc` eslint rules, which require these blocks to be valid.

File: packages/spawner/src/escrow.js (hunk 5 above)
Comment (kriskowal) [reply packages/spawner/src/escrow.js:3]: cc @warner or @michaelfig

File: packages/swingset-runner/demo/resolveChain/bootstrap.js (hunk 6 above)
Comment (kriskowal) [reply packages/swingset-runner/demo/resolveChain/bootstrap.js:12]: This is a style change to appease `eslint`, which insists that the literal be on the right hand side so it doesn’t “sound like Yoda”. It’s debatable. I personally prefer the previous style since the expressions appear in ascending order, but this was the only inconsistency with that rule. I’ll broach the topic.

File: packages/spawner/src/escrow.js (hunk 5 above)
Comment (michaelfig) [reply packages/spawner/src/escrow.js:3]: > Who makes `makePromiseKit` a global? Why?

`spawner` has been mostly unmaintained since its `contractHost` days, which preceded Zoe.  So, it has some peculiarities like this, where it is endowing several "useful globals" to the spawned code.

File: packages/eslint-config/eslint-config.json (hunk 1 above)
Comment (michaelfig) [reply packages/eslint-config/eslint-config.json]: > “I know about that but you shouldn’t use it”

Yes, if this config is omitted, eslint emits a warning (`no-console`) whenever you try to use `console.*`.

**Comment by kriskowal** [comment]
Deferring the creation of an `@agoric/eslint-config` package to another pass since this is a bite-sized victory with passing tests.
